src/AmalgamatorSession.ts

```diff
@@ -706,9 +709,20 @@ export class AmalgamatorSession extends LoggingDebugSession {
                 const childResponse = await childDap.threadsRequest();
                 for (const threadOfChildDap of childResponse.body.threads) {
                     const threadInfo = threadOfChildDap as ThreadInfo;
-                    if (threadInfo.running === false) {
+                    if (
+                        threadInfo.running === false &&
+                        command === 'cdt-amalgamator/resumeAll'
+                    ) {
                         await childDap.continueRequest({ threadId: childId });
                         this.sendEvent(new ContinuedEvent(thread.id, false));
+                    } else if (
+                        threadInfo.running === true &&
+                        command === 'cdt-amalgamator/suspendAll'
+                    ) {
+                        await childDap.pauseRequest({ threadId: childId });
+                        this.sendEvent(
+                            new StoppedEvent('SIGINT', thread.id, false, false)
```

Comment:
I think you should have this as true for `preserveFocusHint`, at least on all but one of them, or else the focus jumps around as each thread is suspended.

```
    /**
     * A value of true hints to the client that this event should not change the
     * focus.
     */
    preserveFocusHint?: boolean;
```

Reply:
Hi @jonahgraham ,
I updated `preserveFocusHint: true`.
Could you please help me review it again?